pkg/config/agentmanagement.go

```diff
@@ -15,9 +15,13 @@ import (
 
 const cacheFilename = "remote-config-cache.yaml"
 
+var defaultConfig = AgentManagement{
+	CacheLocation: "data-agent/",
```

Comment:
This path by default doesn't work  if I try to run the agent locally

Reply:
Furthermore, the agent returns an error but doesn't panic. Polling works but it's not able to save the config file locally

Reply:
`data-agent` can be defined differently based on the command line flags for the wal, so you may want to utilize those command line flags  as the default,`metrics.wal-directory`.

Reply:
Are all these features linux only? IE tmp likely does not exist on windows.

Reply:
Ahh that's a good point, I often forget since I develop on Unix/Linux. Perhaps the best course of action would be to make it a required field, then, if agent management is enabled?

_Or_ the default could be the directory containing the initial config, which would always exist (though may not be writable). In the case that no directory is specified in the `agent_management` block and the directory containing the initial config is not writable, then it would panic.

Reply:
I think there is plenty of good insight on how to do cross-platform persistence in this file https://github.com/grafana/agent/blob/main/pkg/usagestats/reporter.go#L93-L99, where we persist the uid for stats reporting. 

I see 2 options:

* Doing something similar to what we do to persist the uid. I do not like this option because defaulting to something like `/tmp` is dangerous: If the file is gone (tmp cleanup) and the API is down, agents will panic.
* Move the default WAL dir outside of the metrics package into a package with defaults under `config`. That way we can keep the data dir in sync and it is probably a good idea to have a default data folder for the entire agent anyway (and not just for the wal). We could eventually provide defaults for things like `positions_directory` there too.

However, the more I think about it, the more I think we shouldn't provide a default for this field at this point. Not at least until we agree on a default agent data dir we can use globally (changing the default dir later could potentially break existing installations).

What do the others think?

Reply:
Not sure if the chosen `data-agent` path here is chosen for compatibility with Flow's defaults or a coincidence.

For my 2c, gut feeling is the same as Jorge's; I personally would _not_ opt for a default agent dir to be used globally, but have it as a required flag when using agent-management.

Reply:
Flow supports an agent-wide data directory (named `data-agent` by default), so we have one going forward. I'm not sure if it's worth backporting one to the static mode just for agent management; it'd be a fair amount of work.

Reply:
Agree with @jcreixell on making the setting required if Agent Management is turned on.